strategies/failed/src/main/java/org/arquillian/smart/testing/strategies/failed/InProjectTestReporterLoader.java

```diff
@@ -38,11 +47,12 @@
                         }
                     })
                     .map(is -> {
-                        try {
-                            return SurefireReader.loadTestResults(is);
-                        } catch (XMLStreamException e) {
-                            return null;
+                        final Optional<TestResultParser> testResultParser = javaSPILoader.onlyOne(TestResultParser.class, trp -> "junit".equals(trp.type()));
```

Comment:
Well notice that the method comes from JavaSPI which is a generic class, then maybe not in this concrete case but in the future, so what you mean now is that I call the `onlyOne` and that's all isn't it? Yes for now it might has sense as well